client/edge.go

```diff
@@ -53,12 +60,39 @@ type EdgeClient interface {
 
 // NewEdgeConnector create EdgeConnector with the given BrokerConfig for the given EdgeClient
 func NewEdgeConnector(cfg *BrokerConfig, ecl EdgeClient) (*EdgeConnector, error) {
+	var certificates []tls.Certificate
+	var caCertPool *x509.CertPool
+	if len(cfg.DeviceCert) > 0 {
+		keyPair, err := tls.LoadX509KeyPair(cfg.DeviceCert, cfg.DeviceKey)
+		if err != nil {
+			logger.Errorf("error reading client certificate files - \"%s, %s\"", cfg.DeviceCert, cfg.DeviceKey)
+			return nil, err
+		}
+		certificates = []tls.Certificate{keyPair}
+	}
+	if len(cfg.CaCert) > 0 {
+		caCert, err := ioutil.ReadFile(cfg.CaCert)
+		if err != nil {
+			logger.Errorf("error reading CA certificate file - \"%s\"", cfg.CaCert)
+			return nil, err
+		}
+		caCertPool = x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+	}
+	config := &tls.Config{
+		InsecureSkipVerify: false,
+		RootCAs:            caCertPool,
+		Certificates:       certificates,
+		MinVersion:         tls.VersionTLS12,
+		MaxVersion:         tls.VersionTLS13,
+		CipherSuites:       uploaders.SupportedCipherSuites(),
+	}
 	opts := MQTT.NewClientOptions().
 		AddBroker(cfg.Broker).
 		SetClientID(uuid.New().String()).
 		SetKeepAlive(30 * time.Second).
 		SetCleanSession(true).
-		SetAutoReconnect(true)
+		SetAutoReconnect(true).SetTLSConfig(config)
```

Comment:
TLS configuration is always set. Is it correct? What is the behavior in case of plain communication?

Reply:
It also works with plain communication, perhaps config is ignored

Reply:
In general, it's not a good idea to work with "trial and error" approach. This topic has to be clarified.